bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -668,6 +681,83 @@ bool ReplicaImp::relevantMsgForActiveView(const T *msg) {
   }
 }
 
+bool ReplicaImp::validatePreProcessedResultSignatures(preprocessor::PreProcessedResultMsg &msg) {
+  auto [buf, buf_len] = msg.getResultSignaturesBuf();
+  auto sigs = preprocessor::PreProcessResultSignature::deserializeResultSignatureList(buf, buf_len);
```

Comment:
RVO kicks in here so no copy is made. We can't capture the return by reference, because it's a local variable inside deserializeResultSignatureList().

Reply:
OK, I've missed that we return a local variable.